src/ExpireNFTv4.sol

```diff
@@ -0,0 +1,114 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.9;
+
+import "@openzeppelin/contracts/token/ERC721/ERC721.sol";
+import "@openzeppelin/contracts/token/ERC721/extensions/ERC721Enumerable.sol";
+import "@openzeppelin/contracts/token/ERC721/extensions/ERC721Burnable.sol";
+import "@openzeppelin/contracts/access/AccessControl.sol";
+import "@openzeppelin/contracts/utils/Counters.sol";
+import "@openzeppelin/contracts/utils/Base64.sol";
+
+contract ExpireNFTv4 is ERC721, ERC721Enumerable, ERC721Burnable, AccessControl {
+    using Counters for Counters.Counter;
+
+    bytes32 public constant MINTER_ROLE = keccak256("MINTER_ROLE");
+    Counters.Counter private _tokenIdCounter;
+
+    uint256 public constant EXPIRE_TIME_IN_SECONDS = 60 * 60 * 24 * 365;
+    uint256 public constant SUBSCRIPTION_PRICE = 0.10 ether;
```

Comment:
**Future enhancement**: Have you thought of using stablecoins (ERC20) for payments instead of ether?